### PR TITLE
Java arrays are instanceof Object

### PR DESCRIPTION
**Pull request by peterschrammel**

### CODE REVIEW

File: src/java_bytecode/java_bytecode_convert_class.cpp [hunk 1]
```diff
@@ -348,6 +348,12 @@ void java_bytecode_convert_classt::add_array_types(symbol_tablet &symbol_table)
     comp2.set_base_name("data");
     struct_type.components().push_back(comp2);
 
+    irept::subt &bases=
+      struct_type.add(ID_bases).get_sub();
+    irept base(ID_base);
+    base.set(ID_type, symbol_typet("java::java.lang.Object"));
+    bases.push_back(base);
```
Comment (smowton) [inline src/java_bytecode/java_bytecode_convert_class.cpp]: Use `struct_union_typet::baset` and `struct_union_typet::add_base`

**Review by smowton — CHANGES_REQUESTED** [review]

**Comment by thk123** [comment]
A test gen bump would be appreciated for this change.

**Review by thk123 — CHANGES_REQUESTED** [review]
Looks good but TG pointer bump failing

**Comment by peterschrammel** [comment]
@thk123, @smowton, TG is passing now.

**Review by smowton — APPROVED** [review]

File: src/java_bytecode/java_bytecode_convert_class.cpp [hunk 2]
```diff
@@ -478,39 +478,41 @@ void java_bytecode_convert_classt::add_array_types(symbol_tablet &symbol_table)
...
     // we have the base class, java.lang.Object, length and data
     // of appropriate type
-    struct_type.set_tag(symbol_type_identifier);
+    class_type.set_tag(symbol_type_identifier);
 
-    struct_type.components().reserve(3);
-    struct_typet::componentt
-      comp0("@java.lang.Object", symbol_typet("java::java.lang.Object"));
+    class_type.components().reserve(3);
+    class_typet::componentt comp0(
```
Comment (thk123) [inline src/java_bytecode/java_bytecode_convert_class.cpp]: Name suggestion: `base_class_component`

File: src/java_bytecode/java_bytecode_convert_class.cpp [hunk 3]
```diff
@@ -478,39 +478,41 @@ void java_bytecode_convert_classt::add_array_types(symbol_tablet &symbol_table)
...
+      "@java.lang.Object", symbol_typet("java::java.lang.Object"));
     comp0.set_pretty_name("@java.lang.Object");
     comp0.set_base_name("@java.lang.Object");
-    struct_type.components().push_back(comp0);
+    class_type.components().push_back(comp0);
 
-    struct_typet::componentt comp1("length", java_int_type());
+    class_typet::componentt comp1("length", java_int_type());
```
Comment (thk123) [inline src/java_bytecode/java_bytecode_convert_class.cpp]: Name suggestion: `length_component`

File: src/java_bytecode/java_bytecode_convert_class.cpp [hunk 4]
```diff
@@ -478,39 +478,41 @@ void java_bytecode_convert_classt::add_array_types(symbol_tablet &symbol_table)
...
     comp1.set_pretty_name("length");
     comp1.set_base_name("length");
-    struct_type.components().push_back(comp1);
+    class_type.components().push_back(comp1);
 
-    struct_typet::componentt
-      comp2("data", java_reference_type(java_type_from_char(l)));
+    class_typet::componentt comp2(
```
Comment (thk123) [inline src/java_bytecode/java_bytecode_convert_class.cpp]: Name suggestion: `data_component`

**Review by thk123 — APPROVED** [review]

**Pull request merged**